### PR TITLE
Fix an issue around deleting records from the database.

### PR DESCRIPTION
**Pull request by dmandar**

### CODE REVIEW

File: FirebaseRemoteConfig/Tests/Unit/RCNConfigDBManagerTest.m [hunk 1]
```diff
@@ -258,6 +258,89 @@ - (void)testWriteAndLoadMetadataResult {
...
+  // Completion handler for the write to namespace_delete namespace.
+  RCNDBCompletion insertNamespace1Completion = ^void(BOOL success, NSDictionary *result) {
+    XCTAssertTrue(success);
+
+    // Delete the key for given namespace.
+    [_DBManager deleteRecordFromMainTableWithNamespace:namespaceToDelete
+                                      bundleIdentifier:bundleIdentifier
+                                            fromSource:RCNDBSourceActive];
+
+    // check DB read correctly
```
Comment (ryanwilson) [inline FirebaseRemoteConfig/Tests/Unit/RCNConfigDBManagerTest.m]: Nit: Capitalize and add punctuation.

Comment (dmandar) [reply FirebaseRemoteConfig/Tests/Unit/RCNConfigDBManagerTest.m]: Done.

File: FirebaseRemoteConfig/Tests/Unit/RCNConfigDBManagerTest.m [hunk 2]
```diff
@@ -258,6 +258,89 @@ - (void)testWriteAndLoadMetadataResult {
                                }];
 }
 
+// Create a key each for two namespaces, delete it from one namespace, read both namespaces.
+- (void)testDeleteParamAndLoadMainTable {
+  XCTestExpectation *namespaceDeleteExpectation =
+      [self expectationWithDescription:
+                @"Write a key to namespace_delete, delete it and try reading back again."];
```
[the remark this answers is not in the document]
Comment (dmandar) [reply FirebaseRemoteConfig/Tests/Unit/RCNConfigDBManagerTest.m]: Done.

File: FirebaseRemoteConfig/Tests/Unit/RCNConfigDBManagerTest.m [hunk 3]
```diff
@@ -258,6 +258,89 @@ - (void)testWriteAndLoadMetadataResult {
...
+  XCTestExpectation *namespaceKeepExpectation =
+      [self expectationWithDescription:@"Write a key to namespace_keep and read back again."];
+  NSString *namespaceToDelete = @"namespace_delete";
+  NSString *namespaceToKeep = @"namespace_keep";
+  NSString *bundleIdentifier = [NSBundle mainBundle].bundleIdentifier;
```
[the remark this answers is not in the document]
Comment (dmandar) [reply FirebaseRemoteConfig/Tests/Unit/RCNConfigDBManagerTest.m]: Done.

File: FirebaseRemoteConfig/Sources/RCNConfigDBManager.m [hunk 4]
```diff
@@ -909,42 +913,54 @@ - (void)deleteRecordFromMainTableWithNamespace:(NSString *)namespace_p
...
     }
-    [weakSelf executeQuery:SQL];
+    [strongSelf executeQuery:SQL];
   });
 }
 
 - (void)deleteExperimentTableForKey:(NSString *)key {
-  __weak RCNConfigDBManager *weakSelf;
+  __weak RCNConfigDBManager *weakSelf = self;
+  RCNConfigDBManager *strongSelf = weakSelf;
```
[the remark this answers is not in the document]
Comment (dmandar) [reply FirebaseRemoteConfig/Sources/RCNConfigDBManager.m]: Oops.

**Pull request merged**